assets/blocks/reader-registration/index.php

```diff
@@ -63,12 +78,24 @@ function enqueue_scripts() {
  * @param array[] $attrs Block attributes.
  */
 function render_block( $attrs ) {
+	$block_id   = \wp_rand( 0, 99999 );
 	$registered = false;
 	$message    = '';
+
+	/** Setup list subscription */
+	if ( $attrs['newsletterSubscription'] && method_exists( 'Newspack_Newsletters_Subscription', 'get_lists_config' ) ) {
+		$list_config     = \Newspack_Newsletters_Subscription::get_lists_config();
+		$lists           = array_keys( $list_config );
+		$list_map        = array_flip( $lists );
+		$available_lists = array_values( array_intersect( $lists, $attrs['lists'] ) );
+		if ( empty( $available_lists ) ) {
+			$available_lists = [ $lists[0] ];
```

Comment:
If you don't have any lists active in the Engagement wizard, this will throw a PHP warning:

```
PHP Notice:  Undefined offset: 0 in /newspack/app/public/wp-content/plugins/newspack-plugin/assets/blocks/reader-registration/index.php on line 92
```

On the front-end, the same warning will be thrown and the box with the newsletter lists is present but empty:

<img width="1220" alt="Screen Shot 2022-07-18 at 4 04 22 PM" src="https://user-images.githubusercontent.com/2230142/179625182-5438185a-fa5f-439f-ab7c-c1b134d38561.png">

Expected behavior in both cases should be the same as if the "Enable newsletter subscription" option is disabled.

Reply:
Fixed in 43cc118a07d354986a0009eb0dd638cc5d9c1286

Reply:
Editor looks good now, but on the front-end I still see this :ghost: form:

<img width="1215" alt="Screen Shot 2022-07-19 at 4 18 44 PM" src="https://user-images.githubusercontent.com/2230142/179858046-aed30c04-70a6-48f4-b02d-f62c9c85cbf6.png">

Reply:
I was not able to reproduce that. My steps were:

1. Make sure there are lists enabled and with values on the wizard
2. Configure the block with all lists
3. Disable all lists on the engagement wizard
4. Visit rendered block on page:

<img width="633" alt="image" src="https://user-images.githubusercontent.com/820752/179874973-1a09ee58-0431-4014-abf3-cef7ed66cc02.png">

While trying to understand what might have happened in your case, I pushed 8331ec05234dac04f00cc4dc6bbc245c0ee7df43. It forces a default value for the newsletter label in case there's only 1 newsletter selected. It was allowing the label to be left empty.

Reply:
Hmm, when I replicate using your steps now, I see this on the front-end:

<img width="1217" alt="Screen Shot 2022-07-20 at 1 52 15 PM" src="https://user-images.githubusercontent.com/2230142/180069588-86e95b1f-ca5e-4e7f-9c88-8eb1c4b8c4fc.png">

But since there are no lists enabled in the block, what does the checkbox actually do? Shouldn't we hide the whole section if there are no lists selected?

Reply:
I think I see the discrepancy in our tests! Here's what I did:

1. Make sure there are lists enabled with values in the wizard
2. Configure the block with NO lists selected
3. Keep all lists enabled in the Engagement wizard
4. Visit rendered block on page

Reply:
If there are configured lists but none were selected for the block, it'll use the first from the configured lists:

https://github.com/Automattic/newspack-plugin/blob/b210853675231a6f240ec847fd1e0c328ab14c80/assets/blocks/reader-registration/index.php#L111-L113

You should probably be seeing an error notice on your block configuration as well.

Reply:
I know realize that this is counter-intuitive and doesn't reflect the block seen on the editor, 50fc58acab5d6d5ef4e31bba284482278cf35ebb removes this fallback.

Reply:
I get it now, thanks for clarifying. The error message does appear, but in the block editor preview the block doesn't display the newsletters section like it does on the front-end. The render should match in both contexts.

Also, in my mind it's more intuitive to make the block behave as if newsletter signups are disabled in this situation, but that's subjective and therefore non-blocking.

Reply:
:coke:

Reply:
:coke: on the intuition